doc/api-management.md

```diff
@@ -0,0 +1,59 @@
+# API Management
+
+> Guidance below for Theia framework. End products will get better development experience by using private visibility.
+
+Theia is a framework embracing openness, extensibility, and customizability as much as possible:
+- API defaults to public for clients.
+- API defaults to protected for extenders.
+- Language constructions prohibiting runtime access to internals are never used and so on.
+
+Usually, the version management is built around API visibility.
+Particularly, if a public API is broken then the major release is required.
+
+Following the conventional approach is not practicable for Theia since all APIs are more or less public.
+It will slow down API innovation, accumulate technical debt or require many major releases.
+
+Instead, we require the major release only if a stable API is broken.
+Breaking an experimental API can be done with the minor release,
+but not necessary if such API does not have sufficient adoption.
```

Comment:
I don't understand this part. What does "necessary" refer to?

Reply:
i meant if there are no adopters then it can be just removed. I will remove it, it is explained below.